move/patterns/sources/voting.move

```diff
@@ -50,39 +90,50 @@ public fun destroy_for_testing(v: Vote) {
 /// The associated key-ids are [pkg id][vote id].
 public fun create_vote(
     voters: vector<address>,
+    options: u8,
     key_servers: vector<address>,
+    public_keys: vector<vector<u8>>,
     threshold: u8,
     ctx: &mut TxContext,
 ): Vote {
     assert!(threshold <= key_servers.length() as u8);
+    assert!(key_servers.length() == public_keys.length());
     Vote {
         id: object::new(ctx),
         voters,
         key_servers,
+        public_keys,
         threshold,
+        is_finalized: false,
         votes: vector::tabulate!(voters.length(), |_| option::none()),
-        result: option::none(),
+        options,
     }
 }
 
 /// Cast a vote.
 /// The encrypted object should be an encryption of a single u8 and have the senders address as aad.
-public fun cast_vote(vote: &mut Vote, encrypted_vote: EncryptedObject, ctx: &mut TxContext) {
+public fun cast_vote(vote: &mut Vote, encrypted_vote: vector<u8>, ctx: &mut TxContext) {
+    let encrypted_vote = parse_encrypted_object(encrypted_vote);
+
     // The voter id must be put as aad to ensure that an encrypted vote cannot be copied and cast by another voter.
     assert!(encrypted_vote.aad().borrow() == ctx.sender().to_bytes(), EInvalidVote);
 
     // All encrypted vote must have been encrypted using the same key servers and the same threshold.
     // We could allow the order of the key servers to be different, but for the sake of simplicity, we also require the same order.
-    assert!(encrypted_vote.services() == vote.key_servers);
-    assert!(encrypted_vote.threshold() == vote.threshold);
+    assert!(encrypted_vote.services() == vote.key_servers, EInvalidVote);
+    assert!(encrypted_vote.threshold() == vote.threshold, EInvalidVote);
+
+    // Check that the encryptions were created for this vote.
+    assert!(encrypted_vote.id() == vote.id(), EInvalidVote);
+    assert!(encrypted_vote.package_id() == @patterns, EInvalidVote);
```

Comment:
is that a valid code?

Reply:
Yea, it refers to an alias defined in Move.toml.